### PR TITLE
WIP - Updated validation and AJAX response to support Spanish lang

### PR DESCRIPTION
**Pull request by jimmynotjim**
Currently when using the email signup on Spanish pages the validation and AJAX response notifications are in English.

## Changes

- Updated email validator to handle empty form fields
- Updated email validator to respond with Spanish messages if the `.content` container `lang` attr is set to `es`.
- Updated form submission AJAX responses to support Spanish messages if the `.content` container `lang` attr is set to `es`.

## Testing

1. Run `gulp build` and run the project
2. Temp edit https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_includes/organisms/email-signup.html#L34 to include `novalidate` attribute (this is only to test browsers that don't have validation baked in while using a browser that does)
3. Create a page with an email signup and set the language in the config to "Spanish"
4. Test empty, misformed, and correct email addresses in the signup field.

## Screenshots

Empty field
screen shot 2017-08-11 at 3 17 33 pm

Incorrect formatting
screen shot 2017-08-11 at 3 17 42 pm

Signup failed (bad AJAX response)
screen shot 2017-08-11 at 3 17 55 pm

## Notes

- I can't get the AJAX to work locally, is this a known issue?? If so, is there a way to test a successful AJAX response? I'm not convinced the success message will be triggered when lang is `es` due to this condition here https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/unprocessed/js/organisms/FormSubmit.js#L131

## Todos

- The validators could use some refactoring. I've left a few notes, feel free to add more as you see fit.

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/unprocessed/js/modules/util/validators.js [hunk 1]
```diff
@@ -38,16 +41,34 @@ function date( field, currentStatus ) {
...
+  var opts = opts || {};
   var regex =
     '^[a-z0-9\u007F-\uffff!#$%&\'*+\/=?^_`{|}~-]+(?:\\.[a-z0-9' +
     '\u007F-\uffff!#$%&\'*+\/=?^_`{|}~-]+)*@(?:[a-z0-9]' +
     '(?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z]{2,}$';
   var emailRegex = new RegExp( regex, 'i' );
-  if ( field.value && emailRegex.test( field.value ) === false ) {
+  var emptyValidation = empty( field );
+  var isFilled = emptyValidation.required === undefined ?
+                 'true' : emptyValidation.required;
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/modules/util/validators.js]: Does `'true'` need to be a string? Can you make it a boolean?

Comment (jimmynotjim) [inline cfgov/unprocessed/js/modules/util/validators.js]: Oh duh, thanks for catching that.

**Comment by Scotchester** [comment]
Woohoo! Working on the Content server:

screen shot 2017-08-22 at 11 35 41

**Comment by jimmynotjim** [comment]
Thanks all for seeing this through while I was gone

**Pull request merged**